optuna/samplers/_tpe/sampler.py

```diff
@@ -714,7 +714,7 @@ def _split_observation_pairs(
             (infeasible_idx,) = (violation_1d > 0).nonzero()
             assert len(feasible_idx) >= n_below
             feasible_below, feasible_above = _split_observation_pairs(
-                list(np.array(loss_vals)[feasible_idx]), n_below, None
+                [loss_vals[i] for i in feasible_idx], n_below, None
```

Comment:
Without this change, the following warning message is shown:

```bash
/Users/nzw/Documents/optuna> pytest tests/samplers_tests/tpe_tests/test_sampler.py::test_split_observation_pairs_with_constraints_below_all_feasible
===================================================================================== test session starts =====================================================================================
platform darwin -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/nzw/Documents/optuna, configfile: pyproject.toml
collected 1 item

tests/samplers_tests/tpe_tests/test_sampler.py .                                                                                                                                        [100%]

====================================================================================== warnings summary =======================================================================================
tests/samplers_tests/tpe_tests/test_sampler.py::test_split_observation_pairs_with_constraints_below_all_feasible
  /opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/optuna-3.0.0.dev0-py3.9.egg/optuna/samplers/_tpe/sampler.py:717: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
    list(np.array(loss_vals)[feasible_idx]), n_below, None

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Reply:
The change looks reasonable. The current code `list(np.array(loss_vals)[feasible_idx])` returns `List[List[float, List[float]]]` but `_split_observation_pairs` expects `List[Tuple[float, List[float]]]`. So, I think this change also prevents future mypy errors.